WordPress/gutenberg
# [RNMobile] Add 'Insert from URL' option to Video block

## What?
Fixes [wordpress-mobile/gutenberg-mobile#4646](https://github.com/wordpress-mobile/gutenberg-mobile/issues/4646) by adding "Insert from URL" option to the Video block. 

Also addresses [gutenberg-mobile#4941](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4941) as the companion PR.

## Why?
"Insert from URL" is an option for the Audio, Video, and Image blocks on the web, and also for the Audio block (and now, the Image block) on Gutenberg Mobile. This PR adds the same web behavior to the Image block.

## How?
- Adds [onSelectURL handler](https://github.com/WordPress/gutenberg/compare/trunk...derekblank:rnmobile/add-url-option-to-video-block#diff-554d3b870e564544e2ca4958a9aa84bbf084ec9554768371be92c29c74e2b4cfR166-R184) to Video block to handle URL input with the MediaUpload / MediaPlaceholder components.
- Adds [withDispatch handler](https://github.com/WordPress/gutenberg/compare/trunk...derekblank:rnmobile/add-url-option-to-video-block#diff-554d3b870e564544e2ca4958a9aa84bbf084ec9554768371be92c29c74e2b4cfR405-R409) to display error notices to the user.
- Adds [integration with the Embed block](https://github.com/WordPress/gutenberg/compare/trunk...derekblank:rnmobile/add-url-option-to-video-block#diff-554d3b870e564544e2ca4958a9aa84bbf084ec9554768371be92c29c74e2b4cfR171-R177) to handle resources that are embeddable (like YouTube URLs). 

## Testing Instructions
1. Open a Post or Page
2. Insert an Video Block
3. Tap on Insert from URL in the menu options
4. Paste a valid (or invalid) URL into the text input tap Apply
5. Image should display in UI (or show error message if the URL is invalid) 

## Screenshots or screencast 

<details>
  <summary>Embeddable Test Video</summary>

  https://user-images.githubusercontent.com/643285/171572592-d03393e7-3986-4756-b747-641369897973.mov
</details>

<details>
    <summary>Non-Embeddable Test Video</summary>

   https://user-images.githubusercontent.com/643285/171572619-a6068820-d7c8-4844-9241-fbad251e10a3.mov
</details>
